src/patternfly/components/Check/check.scss

```diff
@@ -18,32 +16,27 @@
   --#{$check}__label-required--FontSize: var(--#{$pf-global}--FontSize--sm);
   --#{$check}__label-required--Color: var(--#{$pf-global}--danger-color--100);
 
-  display: grid;
+  display: inline-grid;
   grid-template-columns: auto 1fr;
   grid-gap: var(--#{$check}--GridGap);
+  width: 100%;
```

Comment:
I'm not sure if this is the root cause, but the check in a split button menu toggle is wider than it needs to be (wider than the content) and it's pushing the caret icon over...

What it looks like now
<img width="589" alt="Screenshot 2023-07-14 at 1 09 14 PM" src="https://github.com/patternfly/patternfly/assets/35148959/19823bc4-04ab-47ce-bfa3-9cd445b8d23e">

Without `width: 100%`
<img width="572" alt="Screenshot 2023-07-14 at 1 09 19 PM" src="https://github.com/patternfly/patternfly/assets/35148959/eb87c64f-37a2-47e7-a4df-77960cbe662d">